src/index.js

```diff
@@ -259,11 +259,38 @@ const Route = ({ component, url, matches }) => {
 	return h(component, { url, matches });
 };
 
+class AsyncRoute extends Component {
+	constructor() {
+		super();
+		this.state = {
+			componentData: null
+		};
+	}
+	componentDidMount(){
+		const componentData = this.props.component();
```

Comment:
Random thought (not sure what the best approach to implement it would be): we could cache the promise return value here using the promise-returning function as a key.  That would work really nicely with `bundle-loader?async`.